src/rice2010.jl

```diff
@@ -13,211 +15,144 @@ include("components/slr_component.jl")
-    setparameter(m, :sealeveldamages, :slrdamlinear, slrdamlinear)
-    setparameter(m, :sealeveldamages, :slrdamquadratic, slrdamquadratic)
-
-    connectparameter(m, :sealeveldamages, :TOTALSLR, :sealevelrise, :TOTALSLR)
-    connectparameter(m, :sealeveldamages, :YGROSS, :grosseconomy, :YGROSS)
-
-    # DAMAGES COMPONENT
-    setparameter(m, :damages, :a1, a1)
-    setparameter(m, :damages, :a2, a2)
-    setparameter(m, :damages, :a3, a3)
-
-    connectparameter(m, :damages, :TATM, :climatedynamics, :TATM)
-    connectparameter(m, :damages, :YGROSS, :grosseconomy, :YGROSS)
-    connectparameter(m, :damages, :SLRDAMAGES, :sealeveldamages, :SLRDAMAGES)
-
-    # NET ECONOMY COMPONENT
-    setparameter(m, :neteconomy, :S, savings)
-    setparameter(m, :neteconomy, :l, l)
-
-    connectparameter(m, :neteconomy, :YGROSS, :grosseconomy, :YGROSS)
-    connectparameter(m, :neteconomy, :DAMFRAC, :damages, :DAMFRAC)
-    connectparameter(m, :neteconomy, :DAMAGES, :damages, :DAMAGES)
-    connectparameter(m, :neteconomy, :ABATECOST, :emissions, :ABATECOST)
-
-    # WELFARE COMPONENT
-    setparameter(m, :welfare, :l, l )
-    setparameter(m, :welfare, :elasmu, elasmu )
-    setparameter(m, :welfare, :rr, rr)
-    setparameter(m, :welfare, :scale1, scale1 )
-    setparameter(m, :welfare, :scale2, scale2)
-    setparameter(m, :welfare, :alpha, alpha)
-
-    connectparameter(m, :welfare, :CPC, :neteconomy, :CPC)
-
-    return m
-end
-
-function getrice(;datafile=joinpath(dirname(@__FILE__), "..", "data", "RICE_2010_base_000.xlsm"))
-    params = getrice2010parameters(datafile)
-
-    m = constructrice(params)
-
-    return m
-end
+export RICE
+
+#
+# N.B. See rice2010-defmodel.jl for the @defmodel version of the following
+#
+
+const global datafile = joinpath(dirname(@__FILE__), "..", "data", "RICE_2010_base_000.xlsm")
+p = getrice2010parameters(datafile)
+
+RICE = Model()
+set_dimension!(RICE, :time, 2005:10:2595)
+set_dimension!(RICE, :regions, ["US", "EU", "Japan", "Russia", "Eurasia", "China", "India", "MidEast", "Africa", "LatAm", "OHI", "OthAsia"])
+
+add_comp!(RICE, grosseconomy, :grosseconomy)
+add_comp!(RICE, emissions, :emissions)
+add_comp!(RICE, co2cycle, :co2cycle)
+add_comp!(RICE, radiativeforcing, :radiativeforcing)
+add_comp!(RICE, climatedynamics, :climatedynamics)
+add_comp!(RICE, sealevelrise, :sealevelrise)
+add_comp!(RICE, sealeveldamages, :sealeveldamages)
+add_comp!(RICE, damages, :damages)
+add_comp!(RICE, neteconomy, :neteconomy)
+add_comp!(RICE, welfare, :welfare)
+
+# GROSS ECONOMY COMPONENT
+set_param!(RICE, :grosseconomy, :al, p[:al])
+set_param!(RICE, :grosseconomy, :l, p[:l])
+set_param!(RICE, :grosseconomy, :gama, p[:gama])
+set_param!(RICE, :grosseconomy, :dk, p[:dk])
+set_param!(RICE, :grosseconomy, :k0, p[:k0])
+
+# Note: offset=1 => dependence is on on prior timestep, i.e., not a cycle
+connect_param!(RICE, :grosseconomy, :I, :neteconomy, :I, offset = 1)
```

Comment:
What did we decide about ``offset``? That is not used right now, or is it? If not (i.e. if we are still in a world where the insertion order of components matters), then I think we should remove this here, I'm still not sure I'm convinced of this approach.

Reply:
I'll dig into this further.  Right now if we remove the `offset = 1` we're going to get an error that the graph has a cycle.  The question is whether this is actually a problem with our current implementation, or whether this check is just occurring from previous implementations and could be removed.  We likely need input from @rjplevin on this one.